importers/djornl/parser.py

```diff
@@ -28,161 +30,371 @@ def _configure(self):
         configuration = config.load_from_env(extra_required=['ROOT_DATA_PATH'])
 
         # Collection name config
-        configuration['_NODE_NAME'] = 'djornl_node'
-        configuration['_EDGE_NAME'] = 'djornl_edge'
+        configuration['node_name'] = 'djornl_node'
+        configuration['edge_name'] = 'djornl_edge'
 
-        # Path config
-        configuration['_NODE_PATH'] = os.path.join(
-            configuration['ROOT_DATA_PATH'],
-            'aranet2-aragwas-MERGED-AMW-v2_091319_nodeTable.csv'
-        )
-        configuration['_NODE_FILE_COL_COUNT'] = 20
+        # fetch the manifest and make sure all the files listed actually exist
+        manifest = self._get_manifest(configuration)
+        for type in ['node', 'edge', 'cluster']:
+            configuration[type + '_files'] = []
+
+        error_list = []
+        for file in manifest['file_list']:
+            file_path = os.path.join(configuration['ROOT_DATA_PATH'], file['path'])
+
+            if not os.path.exists(file_path):
+                error_list.append(f"{file_path}: file does not exist")
+                continue
+
+            if not os.path.isfile(file_path):
+                error_list.append(f"{file_path}: not a file")
+                continue
+
+            # add the file to the appropriate list
+            file['file_path'] = file_path
+            configuration[file['data_type'] + '_files'].append(file)
+
+        if error_list:
+            raise RuntimeError("\n".join(error_list))
 
-        configuration['_EDGE_PATH'] = os.path.join(
-            configuration['ROOT_DATA_PATH'],
-            'merged_edges-AMW-060820_AF.tsv'
-        )
-        configuration['_EDGE_FILE_COL_COUNT'] = 5
-
-        _CLUSTER_BASE = os.path.join(configuration['ROOT_DATA_PATH'], 'cluster_data')
-        configuration['_CLUSTER_PATHS'] = {
-            'markov_i2': os.path.join(
-                _CLUSTER_BASE,
-                'out.aranetv2_subnet_AT-CX_top10percent_anno_AF_082919.abc.I2_named.tsv'
-            ),
-            'markov_i4': os.path.join(
-                _CLUSTER_BASE,
-                'out.aranetv2_subnet_AT-CX_top10percent_anno_AF_082919.abc.I4_named.tsv'
-            ),
-            'markov_i6': os.path.join(
-                _CLUSTER_BASE,
-                'out.aranetv2_subnet_AT-CX_top10percent_anno_AF_082919.abc.I6_named.tsv'
-            ),
-        }
         self._config = configuration
         return self._config
 
+    def _get_manifest_schema_file(self):
+
+        return os.path.join(self._get_dataset_schema_dir(), 'manifest.schema.json')
+
+    def _get_dataset_schema_dir(self):
+
+        if not hasattr(self, '_dataset_schema_dir'):
+            dir_path = os.path.dirname(os.path.realpath(__file__))
+            self._dataset_schema_dir = os.path.join(dir_path, '../', '../', 'spec', 'datasets', 'djornl')
```

Comment:
You don't have to change anything, but it may be easier to just use the current working directory and assume/assert that it is always the root of the project

Reply:
That would break the tests (and other use cases). This ensures that you'll get the correct dir regardless of where you're calling from.

Reply:
You have to cd within the project to run certain tests?